common/xml/fpga_architecture.xsd

```diff
@@ -1,836 +1,873 @@
-        <xs:complexType>
-          <xs:attribute name="type" type="func_type" use="required" />
-          <xs:attribute name="formula" type="xs:string" use="required" />
-        </xs:complexType>
-      </xs:element>
-    </xs:sequence>
-  </xs:complexType>
-  
-  <xs:complexType name="wireconn_connection">
-    <xs:attribute name="type" type="xs:string" use="required"/>
-    <xs:attribute name="switchpoint" type="xs:string" use="required"/>
-  </xs:complexType>
-  
-  <xs:complexType name="wireconn">
-    <xs:choice minOccurs="0" maxOccurs="unbounded">
-      <xs:element name="from" type="wireconn_connection" maxOccurs="unbounded"/>
-      <xs:element name="to" type="wireconn_connection" maxOccurs="unbounded"/>
-    </xs:choice>
-    <xs:attribute name="num_conns_type" type="xs:string" use="required"/>
-    <xs:attribute name="from_type" type="xs:string"/>
-    <xs:attribute name="to_type" type="xs:string"/>
-    <xs:attribute name="from_switchpoint" type="xs:string"/>
-    <xs:attribute name="to_switchpoint" type="xs:string"/>
-  </xs:complexType>
-
-  <xs:complexType name="switchblock">
-    <xs:sequence>
-      <xs:element name="switchblock_location" type="switchblock_location" maxOccurs="unbounded"/>
-      <xs:element name="switchfuncs" type="switchblock_func" maxOccurs="unbounded"/>
-      <xs:element name="wireconn" type="wireconn" maxOccurs="unbounded"/>
-    </xs:sequence>
-    <xs:attribute name="name" type="xs:string" use="required"/>
-    <xs:attribute name="type" type="switchblock_type" use="required"/>
-  </xs:complexType>
-
-  <xs:complexType name="switchblocklist">
-    <xs:sequence>
-      <xs:element name="switchblock" type="switchblock" maxOccurs="unbounded"/>
-    </xs:sequence>
-  </xs:complexType>
-
-  <xs:complexType name="clock">
-    <xs:attribute name="C_wire" type="xs:double"/>
-    <xs:attribute name="C_wire_per_m" type="xs:double"/>
-    <xs:attribute name="buffer_size" type="buf_size"/>
-  </xs:complexType>
-
-  <xs:complexType name="clocks">
-    <xs:sequence>
-      <xs:element name="clock" type="clock" minOccurs="0" maxOccurs="unbounded"/>
-    </xs:sequence>
-  </xs:complexType>
-
-  <xs:element name="architecture">
-    <xs:complexType>
-      <xs:choice minOccurs="0" maxOccurs="unbounded">
-        <xs:element name="models" type="models"/>
-        <xs:element name="layout" type="layout"/>
-        <xs:element name="device" type="device"/>
-        <xs:element name="switchlist" type="switchlist"/>
-        <xs:element name="switchblocklist" type="switchblocklist"/>
-        <xs:element name="segmentlist" type="segmentlist"/>
-        <xs:element name="power" type="global_power"/>
-        <xs:element name="clocks" type="clocks"/>
-        <xs:element name="directlist" type="directlist"/>
-        <xs:element name="tiles" type="tiles"/>
-        <xs:element name="complexblocklist" type="complexblocklist"/>
-      </xs:choice>
-    </xs:complexType>
-  </xs:element>
-
-</xs:schema>
+<?xml version="1.0"?>
```

Comment:
Why is every single line modified?

Reply:
Old schema had CRLF, mine has LF.

Reply:
Considering the rest of the repository uses LF newlines, maybe I shouldn't fix this? But it would be good to divide into two commits, first newline changes, then real changes.

Reply:
Please don't change line endings.